src/app/editor/code/editor-container/codegen/xamarin-forms/codegen/shape.service.ts

```diff
@@ -0,0 +1,212 @@
+// class to recognise if sketchapp shape is a rectangle, a line, a circle(TODO) or something else
+
+class Point {
+    x: number;
+    y: number;
+
+    constructor(strOrX: string | number, y: number = null) {
+        // a string containing the coords is input
+        if (y === null) {
+            const coords: any = this.stringToCoords(strOrX as string);
+            if (coords) {
+                this.x = this.decRound(coords.x);
+                this.y = this.decRound(coords.y);
+            }
+            // the coords numbers are input
+        } else {
+            this.x = this.decRound(strOrX as number);
+            this.y = this.decRound(y);
+        }
+    }
+
+    // 2 decimals rounding
+    decRound(number: number): number {
+        return Math.round(number * 100 + 0.001) / 100;
+    }
+
+    distanceRounded(point: Point): number {
+        return this.decRound(this.distance(point));
+    }
+    distance(point: Point): number {
+        return Math.sqrt(this.distanceSquared(point));
+    }
+    distanceSquaredRounded(point: Point): number {
+        return this.decRound(this.distanceSquared(point));
+    }
+    distanceSquared(point: Point): number {
+        return Math.pow(this.x - point.x, 2) + Math.pow(this.y - point.y, 2);
+    }
+
+    stringToCoords(string: string): RegExpMatchArray['groups'] | boolean {
+        const regex = /{(?<x>\-?\d(?:\.\d+)?(?:e\-?\d+)?),\s?(?<y>\-?\d(?:\.\d+)?(?:e\-?\d+)?)}/;
+        const match = string.match(regex);
+        if (!!match) {
+            return match.groups;
+        }
+        return false;
```

Comment:
We don't seem to use the matched groups of this regex, in the rest of this PR. We could then just run a test and return a boolean. WDYT?

```suggestion
        const regex = /{(?<x>\-?\d(?:\.\d+)?(?:e\-?\d+)?),\s?(?<y>\-?\d(?:\.\d+)?(?:e\-?\d+)?)}/;
        return regex.test(string);
```

Reply:
This function is used to parse a string of the form `"{1, 2}"` into an object of the form `{x: 1, y: 2}`
It either return this object or false if the parsing failed
We need this object in the constructor of the Point class

Reply:
nvm.